dpppt-backend-sdk/dpppt-backend-sdk-model/src/main/java/org/dpppt/backend/sdk/utils/UTCInstant.java

```diff
@@ -0,0 +1,184 @@
+    }
+    public OffsetDateTime getOffsetDateTime() {
+        return OffsetDateTime.ofInstant(getInstant(), ZoneOffset.UTC);
+    }
+    public LocalDateTime getLocalDateTime() {
+        return LocalDateTime.ofInstant(getInstant(), ZoneOffset.UTC);
+    }
+    public UTCInstant atStartOfDay() {
+        return new UTCInstant(getLocalDate().atStartOfDay().toInstant(ZoneOffset.UTC).toEpochMilli());
+    }
+    public LocalDate getLocalDate() {
+        return getLocalDateTime().toLocalDate();
+    }
+    public LocalTime getLocalTime() {
+        return getLocalDateTime().toLocalTime();
+    }
+
+    public UTCInstant plus(Duration duration){
+        return new UTCInstant(this.timestamp + duration.toMillis());
+    } 
+    public UTCInstant minus(Duration duration){
+        return new UTCInstant(this.timestamp - duration.toMillis());
+    } 
+    //TODO: Fix for leap year and stuff
+    public UTCInstant plusYears(long years){
+        return new UTCInstant(this.timestamp + Duration.ofDays(years * 365).toMillis());
+    } 
+    //TODO: Fix for leap year and stuff
+    public UTCInstant minusYears(long years){
+        return new UTCInstant(this.timestamp - Duration.ofDays(years * 365).toMillis());
+    } 
+    public UTCInstant plusDays(long days){
+        return new UTCInstant(this.timestamp + Duration.ofDays(days).toMillis());
+    } 
+    public UTCInstant minusDays(long days){
+        return new UTCInstant(this.timestamp - Duration.ofDays(days).toMillis());
+    } 
+    public UTCInstant plusHours(long hours){
+        return new UTCInstant(this.timestamp + Duration.ofHours(hours).toMillis());
+    } 
+    public UTCInstant minusHours(long hours){
+        return new UTCInstant(this.timestamp - Duration.ofHours(hours).toMillis());
+    } 
+    public UTCInstant plusMinutes(long minutes){
+        return new UTCInstant(this.timestamp + Duration.ofMinutes(minutes).toMillis());
+    } 
+    public UTCInstant minusMinutes(long minutes){
+        return new UTCInstant(this.timestamp - Duration.ofMinutes(minutes).toMillis());
+    } 
+    public UTCInstant plusSeconds(long seconds){
+        return new UTCInstant(this.timestamp + Duration.ofSeconds(seconds).toMillis());
+    } 
+    public UTCInstant minusSeconds(long seconds){
+        return new UTCInstant(this.timestamp - Duration.ofSeconds(seconds).toMillis());
+    } 
+
+    public boolean isMidnight() {
+        return getLocalTime().equals(LocalTime.MIDNIGHT);
+    }
+    public long getTimestamp() {
+        return timestamp;
+    }
+    public long get10MinutesSince1970() {
+        return getDuration(midnight1970()).dividedBy(GaenUnit.TenMinutes.getDuration());
+    }
+
+
+
+    public boolean isSameDate(UTCInstant other) {
+        return this.getLocalDate().isEqual(other.getLocalDate());
+    }
+    public boolean isSameTime(UTCInstant other) {
+        return this.getLocalTime().equals(other.getLocalTime());
+    }
+
+    public boolean isBeforeExact(UTCInstant other) {
```

Comment:
These and the other `is` are very important, this is why I would like them to be as intuitive as possible. Currently they confuse me...
- `isBeforeExact` - because it's the 'exact' time?
- `isBeforeTime` - because it's the `TimeOfDay`?
- `is(Before|After)Today` - IMO they should not be here - they re-create a `now()`, which is one thing we should avoid...

My proposition:
- `is(Before|After)` - compares using timestamp - overloaded for `UTCInstant`, `long`, `LocalDate`
- `is(Before|After)Date` - compares using `LocalDate` - overloaded for the same, or perhaps no overloading, to make it clear that it's a `LocalDate`
- `is(Before|After)TimeOfDay` - compares using `LocalTime` - no overloading

Reply:
I adjusted the naming to be more explicit. But actually I want to avoid having too many overloads, since this then hides logic in the `UTCInstant` class again, and I tried to  have something as expressive as possible. I pushed the new suggestion :)